### ISSUE TITLE
a minor bug of escalateEvent

### ISSUE THREAD

**Issue by gengshenghong** [jmoenig/Snap--Build-Your-Own-Blocks#1159]
See below a fix of a minor bug (I think :) ) of escalateEvent:

Morph.prototype.escalateEvent = function (functionName, arg) {
    var handler = this.parent;
    if (handler !== null) { // if morph's parent is null, ignore the event.
        while (!handler[functionName] && handler.parent !== null) {
            handler = handler.parent;
        }
        if (&& handler[functionName]) {
            handler[functionName](arg);
        }
    }
};

I get this crash in snap when invoking escalateEvent myself, it seems the Morph's parent is null and cause the crash. The escalateEvent should be fixed to handle this case (or all code invoking escalateEvent has to do the validation instead).

Thanks,
Shenghong

**Comment by brollb** [comment]
I believe you accidently made a markdown link in your comment. Put your code in code blocks with 
```
\```
[code goes here]
\```
```

Otherwise you can use an escaping backslash with
``` 
\[asdf]()
```

**Comment by jmoenig** [comment]
I'd like to know how this ever happens? As soon as a Morph is "in the World" it can handle (and escalate) events, the issue you're trying to fix can only happen to off-world Morphs where raising an events doesn't make sense to me...

**Comment by gengshenghong** [comment]
@jmoenig  I agree with you. I was doing some exploration on the source and add escalateEvent to the "InputField" morph (I want to see whether it is possible to fire an event when a block's parameters in the script area changed but finally failed)  and cause the crash. I am also confused why it does not have a parent (off-world morph)....as there are too many InputField in the "world", I just added above code to avoid the crash.
